api/views.py

```diff
@@ -40,31 +40,30 @@ def post(self, request):
         Return a query set according to the post message status.
         """
         try:
-            request_type = request.data['type']
-            size = request.data['limit']
+            request_type = request.data.get('type')
+            size = request.data.get('limit', None)
 
             if request_type == "brownbag":
-                # Get all users IDs elligible for brownbag
-                users_queryset = User.objects.filter(
-                    profile__brownbag="not_done").values_list('id', flat=True)
-                users = list(users_queryset)
-                rand = Randomizer(users)
-                next_brownbag_user = rand.get_random()
-                today = datetime.date.today()
-                weekday = today.weekday()
-                next_friday = today + datetime.timedelta((4 - weekday) % 7)
-                data = {
-                    "date": str(next_friday),
-                    "status": "next_in_line",
-                    "user": next_brownbag_user
-                }
-                serializer = BrownbagSerializer(data=data)
-                if serializer.is_valid():
-                    serializer.save()
+                try:
```

Comment:
The second try is there specifically to handle the usecase of creating a Brownbag twice. Since the date (in Brownbag model) is unique, the execution will shift to the IntegrityError and from there, we can create a new one but with a different Friday date.